src/Sparrow/Collections/RingBuffer.cs

```diff
@@ -50,25 +50,23 @@ public SingleConsumerRingBuffer(int size)
         public bool TryPush(ref T item)
         {
             // We will check if we potentially have the ability to push 1 more item. 
-            int cidx = Volatile.Read(ref _currentIdx);
             int sidx = Volatile.Read(ref _startIdx);
 
+            //_currentIdx modified only by producer and there for no need of Volatile.Read
             // Check if we have empty spaces in the buffer.
-            if (cidx - sidx >= _size)
+            if (_currentIdx - sidx >= _size)
                 return false; // No space to do anything
 
-            int ticker = Interlocked.Increment(ref _currentIdx) - 1;
-
             // Assign the value
-            ref var cl = ref _buffer[ticker & _mask];
-
+            ref var cl = ref _buffer[_currentIdx & _mask];
             // We assign the item
             cl.Item = item;
```

Comment:
The write and the read to the array are not by `Volatile` except for the writing to `RingItem<T>.IsReady.
Is it a problem?

Reply:
@redknightlois Can you take a look?